README.rst

```diff
@@ -26,4 +26,28 @@ the output files.
         -q, --quiet            Be quiet, be vewy vewy quiet
         -V, --version          Report installed version
         -z, --compress         Ccompress (gzip) the target(s)
-        -b, --blocksize [int]  Maximum size of pre-compressed output files in bytes. (default: 1048576)
+        $-b, --blocksize [int]  Maximum size of pre-compressed output files in bytes. (default: 1048576)
+
+
+Architecture
+============
+
+s3lncoll has a pipe and filter architecture which streams a bucket of
+keys through a `LineStream`. The bucket of keys are opened and `LineStream`
+spits out a single line using an iterator. `RotatingFileCtx` aggregates these
+lines into fixed chuck size files and flushes the file into S3 after gnuzipping 
+the files into a different s3 bucket location.
```

Comment:
Into files of a maximum size or that of a single line, whichever is the larger.

It doesn't have to be a different S3 location.  It could be in place.